### PR TITLE
feat: add pre agg meta table

### LINKED ISSUE #1197: Pre-aggregator Metatable initialization
**Is your feature request related to a problem? Please describe.**
sub-task of support long-time window feature

**Describe the solution you'd like**
The creation/initialization  of the metatable will be done during system start, as other system tables.
It is an internal table which is not visible to normal users.

**Additional context**
long-time window feature

### PR DESCRIPTION
**Pull request by nautaa**
close #1197 

add pre agg meta table.
sub-task of support long-time window feature.

### CODE REVIEW

**Comment by tobegit3hub** [comment]
We should add new branch for `0.4` so that these new features only be merged in `0.5` and later.

**Comment by zhanghaohit** [comment]
[long time window - compute engine](https://trello.com/c/XMMF5TXl/6-long-time-window-compute-engine)

File: src/nameserver/system_table.h [hunk 1]
```diff
@@ -64,6 +66,23 @@ class SystemTable {
...
                 break;
             }
+            case SystemTableType::KPreAggMetaInfo: {
+                SetColumnDesc("aggr_table", openmldb::type::DataType::kString, table_info->add_column_desc());
+                SetColumnDesc("base_db", openmldb::type::DataType::kString, table_info->add_column_desc());
+                SetColumnDesc("base_table", openmldb::type::DataType::kString, table_info->add_column_desc());
+                SetColumnDesc("aggr_func", openmldb::type::DataType::kString, table_info->add_column_desc());
+                SetColumnDesc("aggr_col", openmldb::type::DataType::kString, table_info->add_column_desc());
+                SetColumnDesc("partition_cols", openmldb::type::DataType::kString, table_info->add_column_desc());
+                SetColumnDesc("order_by_col", openmldb::type::DataType::kString, table_info->add_column_desc());
```
Comment (zhanghaohit) [inline src/nameserver/system_table.h:76]: also add an index for cols (base_db, base_table, partition_cols, order_by_col, aggr_func, aggr_col), which can uniquely identify an `aggr_table`

File: src/nameserver/system_table_test.cc [hunk 2]
```diff
@@ -60,9 +60,14 @@ TEST_F(SystemTableTest, SystemTable) {
     ns_client.Init();
     std::vector<::openmldb::nameserver::TableInfo> tables;
     std::string msg;
-    ASSERT_TRUE(ns_client.ShowTable("", INTERNAL_DB, false, tables, msg));
```
Comment (zhanghaohit) [inline src/nameserver/system_table_test.cc:63]: keep the "empty" name case. In our current setting, it will be get two system tables.

**Review by zhanghaohit — CHANGES_REQUESTED** [review]

File: src/nameserver/system_table.h [hunk 3]
```diff
@@ -76,11 +76,24 @@ class SystemTable {
...
+                
+                auto index2 = table_info->add_column_key();
+                index2->set_index_name("unique_key");
+                index2->add_col_name("base_db");
+                index2->add_col_name("base_table");
+                index2->add_col_name("aggr_func");
+                index2->add_col_name("aggr_col");
+                index2->add_col_name("partition_cols");
+                index2->add_col_name("order_by_col");
+                index2->add_col_name("bucket_size");
```
Comment (zhanghaohit) [inline src/nameserver/system_table.h]: `bucket_size` is not necessary

File: src/nameserver/system_table.h (hunk 1 above)
Comment (nautaa) [reply src/nameserver/system_table.h:76]: We should also add `bucket size` to the index columns to uniquely identify a pre-aggregated table.

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Review by dl239 — APPROVED** [review]

**Pull request merged**